pkg/net/round_tripper.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022 The KCP Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package net
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
```

Comment:
I think we can avoid bringing this dependency by using the std lib clone for requests

```
req = utilnet.CloneRequest(req)
```

by
```
req := req.Clone(req.Context())
```

Reply:
it does look like they should be pretty much equivalent